src/engine/include/includejs/engine_private_data.h

```diff
@@ -0,0 +1,32 @@
+#ifndef INCLUDEJS_ENGINE_PRIVATE_DATA_H
+#define INCLUDEJS_ENGINE_PRIVATE_DATA_H
+
+#include <includejs/engine_value.h>
+
+#include <map>    // std::map
+#include <memory> // std::unique_ptr
+
+namespace includejs {
+
+// Inspired by https://github.com/sourcemeta/jsontoolkit
+/// @ingroup engine
+struct INCLUDEJS_ENGINE_EXPORT PrivateObjectData {
+  std::unique_ptr<std::map<void *, Value::Function>> functions =
```

Comment:
FWIW, this is only useful for the JavaScriptCore implementation, so the unique_ptr should be a part of the `data` that's being stored, not a member of this class.

Reply:
Yeah, it makes sense, but I expect that you'd finally do the same instead of storing the function into the global object.

But even with this, now that I think about it, Imagine that you would like to store function into a vector for v8 instead of a map; this class should support such a change.

Now the question is, do we want to uniformize this between engines? Because that means that we'll have to create another class that will contain:
- the raw data files, (void *)
- the function store

I wonder if finally, we should not have a class of PrivateObjectData by engine.

Reply:
> but I expect that you'd finally do the same for v8 instead of storing the function into the global object.

Why?

> Imagine that you would like to store function into a vector for v8 instead of a map; this class should support such a change.

Why would that be necessary?

Reply:
> Why?

Last time during the call, you mentioned this being UB, no?

> Why would that be necessary?

Dunno, for whatever reason.

Anyway, the whole purpose of this class is to contain a pointer and a function store.

So maybe we split like that:

- a class that takes care of cleaning the private field (just that
- another class that helps us to abstract what we need to store precisely (function store for instance)

Reply:
> a class that takes care of cleaning the private field (just that

But even that, do we are really sure that we'll need this everywhere.

Having a full PrivateDataClass by engine is prolly the simplest solution to start

Reply:
Also, feel free to share your potential solution :)